app/src/main/java/org/dhis2/data/forms/dataentry/FormView.kt

```diff
@@ -187,6 +169,26 @@ class FormView(
     var scrollCallback: ((Boolean) -> Unit)? = null
     private var displayConfErrors = true
     private var onSavePicture: ((String) -> Unit)? = null
+    private lateinit var configuration: FormViewConfiguration
+
+    override fun onAttach(context: Context) {
+        super.onAttach(context)
+        when {
+            context is FormViewConfigurationProvider -> {
+                this.configuration = (context as FormViewConfigurationProvider).configuration()
+            }
+            parentFragment is FormViewConfigurationProvider -> {
+                this.configuration =
+                    (parentFragment as FormViewConfigurationProvider).configuration()
+            }
+            else -> {
+                throw RuntimeException(
+                    "Activity or parent fragment must implement the " +
+                        "FormViewConfigurationProvider"
+                )
+            }
+        }
+    }
```

Comment:
I think the general idea of the configurator is good. Still, why attached to android activity?? we are imposing a restriction to everyone who wants to use the form to have a configurator in the activity that does not seem to be linked anywhere since you catch it with (context as // parent ass).configurator.

We have many options for example A think a simpler one might be returning a builder for the fragment and setting every listener in a similar way as before, you would not have a configurator but since it does not have any "logic" i am not sure about its purpose, the same could be done for the remaining fields. Thoughts @Balcan  @andresmr @ferdyrod  @mmmateos ???